automem/dedup.py

```
@@ -0,0 +1,184 @@
+    try:
+        embedding = generate_embedding(new_content)
+    except Exception:
+        logger.warning("Failed to generate embedding for dedup check, defaulting to ADD")
+        return result
+
+    if not embedding:
+        return result
+
+    # Step 2: Search for similar existing memories
+    try:
+        from qdrant_client.models import Filter  # noqa: F401
```

Ruff (0.15.1)

[warning] Unused `noqa` directive (non-enabled: `F401`)

Remove unused `noqa` directive

(RUF100)